### PR TITLE
refactor(common): Log a warning when the priority attribute of `NgOpt…

### PR DESCRIPTION
**Pull request by JeanMeche**
…imizedImage` is used too often.

When the DOM content is loaded, Angular will log a warning message if the `priority` attribute is applied to often on `NgOptimizedImage` directive instances.

### CODE REVIEW

File: packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts [hunk 1]
```diff
@@ -1245,6 +1265,27 @@ function assertNoLoaderParamsWithoutLoader(dir: NgOptimizedImage, imageLoader: I
...
+function assetPriorityCountBelowThreshold() {
+  if (PRIORITY_TRUE_COUNT === 0) {
+    document.addEventListener('DOMContentLoaded', (event) => {
+      if (PRIORITY_TRUE_COUNT > PRIORITY_COUNT_THRESHOLD) {
+        console.warn(
+          formatRuntimeError(
+            RuntimeErrorCode.TOO_MANY_PRIORITY_ATTRIBUTES,
+            `NgOptimizedImage: The "priority" attribute is set to true more than ${PRIORITY_COUNT_THRESHOLD} times. ` +
+              `Having too many high priority images will slow down LCP. ` +
+              `Make sure to only apply the priority attribute on the images that affect LCP the most.`,
```
Comment (kara) [inline packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts:1280]: Added a bit more detail here but I'm wondering if maybe we want to have an FAQ item in the docs as well to explain how to find your LCP image with Chrome Dev Tools and what to do if you want to disable lazy loading without over-boosting priority. (Should probably be a follow-up).

```suggestion
            `NgOptimizedImage: The "priority" attribute is set to true more than ${PRIORITY_COUNT_THRESHOLD} times (${PRIORITY_TRUE_COUNT} times). ` +
              `Marking too many images as "high" priority can hurt your application's LCP (https://web.dev/lcp).` +
              `"Priority" should only be set on the image expected to be the page's LCP element.`,
```

**Review by kara — APPROVED** [review]
LGTM!

**Review by AndrewKushnir — COMMENTED** [review]
@JeanMeche thanks for adding this check 👍

File: packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts (hunk 1 above)
Comment (JeanMeche) [reply packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts:1280]: Sure, I'll add this to my todo list.

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:647011157:BASE:647011191:1719425313405:6d1a48c9).

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
**Caretaker notes**:

- the presubmit is "green"
- the only change that requires the `public-api` approval is a new error code, so it should be safe to proceed with the merge with the current approvals

**Review by dylhunn — APPROVED** [review]

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit cc4cb9608a916d63bdefd544ecd125e2f70eb062.

The changes were merged into the following branches: main

**Pull request closed without merging**